lib/rucio/db/sqla/util.py

```diff
@@ -514,3 +513,31 @@ def temp_table_mngr(session: "Session") -> TempTableManager:
         mngr = TempTableManager(session)
         session.info[key] = mngr
     return mngr
+
+
+def result_to_dict(res: Result) -> Dict[str, Any]:
+    """
+    Constructs a dictionary from a sqlalchemy result.
+
+    `dict(res)` would be correct too, however, sqlalchemy adds it's own keys
+    (for now `_sa_instance_state`). To not add unnecessary keys, we iterate over
+    the expected ones and add them to the result.
+
+    :param res: One result of a sqlalchemy query.
+    :returns: A dicitonary with the items of the res parameter.
+    """
+    dic = {}
+    for column in res.__table__.columns:
```

Comment:
did you check if https://docs.sqlalchemy.org/en/14/core/connections.html#sqlalchemy.engine.Row._asdict 

can be doing exactly what we want ?

Reply:
Yes, if we can trust StackOverflow there are multiple methods of doing this. The `_asdict` only works on queries with selected fields (again, source is StackOverflow): https://stackoverflow.com/a/21029850

This is the reason why I'm introcuding the funtion and don't just use an already provided method...

Reply:
Just went deeper into it and you are right.